### PR TITLE
Add work-around to sleet-probability code

### PR DESCRIPTION
**Pull request by MoseleyS**
Adds optional argument to demote snow-rain mismatch errors to warnings and sets sleet to zero in these cases so that we can run the new sleet-chains while a separate investigation is undertaken to fix the underlying science.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

File: improver/calculate_sleet_prob.py [hunk 1]
```diff
@@ -31,13 +31,15 @@
 """A plugin to calculate probability of sleet"""
 
 import numpy as np
+import warnings
```
Comment (benfitzpatrick) [inline improver/calculate_sleet_prob.py]: Codacy Issue found: [standard import "import warnings" should be placed before "import numpy as np"](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4880834)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- improver_tests/acceptance/test_sleet_probability.py  1
- improver_tests/calculate_sleet_prob/test_calculate_sleet_probability.py  1
- improver/calculate_sleet_prob.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4880834&bid=16173889)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/acceptance/test_sleet_probability.py  1
- improver_tests/calculate_sleet_prob/test_calculate_sleet_probability.py  1
- improver/calculate_sleet_prob.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4880834&bid=16173889)

File: improver/cli/sleet_probability.py [hunk 2]
```diff
@@ -37,7 +37,9 @@
 @cli.clizefy
 @cli.with_output
 def process(snow: cli.inputcube,
-            rain: cli.inputcube):
+            rain: cli.inputcube,
+            *,
+            ignore_mismatch: bool = False):
```
Comment (neilCrosswaite) [inline improver/cli/sleet_probability.py]:
```suggestion
            ignore_mismatch=False):
```
Thi should work with clize fine.

Comment (MoseleyS) [reply improver/cli/sleet_probability.py]: Changed.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/acceptance/test_sleet_probability.py  1
- improver_tests/calculate_sleet_prob/test_calculate_sleet_probability.py  1
- improver/calculate_sleet_prob.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4880834&bid=16173889)

**Pull request merged**